src/Workspaces/Core/Portable/Log/RoslynEventSource.LogBlock.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Internal.Log
+{
+    internal partial class RoslynEventSource
+    {
+        /// <summary>
+        /// Logs an informational block with given <paramref name="entity"/>'s <see cref="object.ToString"/> representation as the message
+        /// and specified <paramref name="functionId"/>.
+        /// On dispose of the returned disposable object, it logs the 'tick' count between the start and end of the block.
+        /// Unlike other logging methods on <see cref="RoslynEventSource"/>, this method does not check
+        /// if the specified <paramref name="functionId"/> was explicitly enabled.
+        /// Instead it checks if the <see cref="RoslynEventSource"/> was enabled at <see cref="EventLevel.Informational"/> level.
+        /// </summary>
+        public static IDisposable LogInformationalBlock(FunctionId functionId, object entity, CancellationToken cancellationToken)
+            => LogBlock.Create(functionId, entity, EventLevel.Informational, cancellationToken);
+
+        /// <summary>
+        /// Logs an informational message block with the given <paramref name="message"/>> and specified <paramref name="functionId"/>.
+        /// On dispose of the returned disposable object, it logs the 'tick' count between the start and end of the block.
+        /// Unlike other logging methods on <see cref="RoslynEventSource"/>, this method does not check
+        /// if the specified <paramref name="functionId"/> was explicitly enabled.
+        /// Instead it checks if the <see cref="RoslynEventSource"/> was enabled at <see cref="EventLevel.Informational"/> level.
+        /// </summary>
+        public static IDisposable LogInformationalBlock(FunctionId functionId, string message, CancellationToken cancellationToken)
+            => LogBlock.Create(functionId, message, EventLevel.Informational, cancellationToken);
+
+        /// <summary>
+        /// This tracks the logged message. On instantiation, it logs 'Started block' with other event data.
+        /// On dispose, it logs 'Ended block' with the same event data so we can track which block started and ended when looking at logs.
+        /// </summary>
+        private struct LogBlock : IDisposable
```

Comment:
Why the struct if we're still going to box it because the other APIs return IDisposable directly?

Reply:
@sharwell any objections to making it a class again?

Reply:
I expected the APIs to return the struct